command/agent/config/config.go

```diff
@@ -260,6 +266,25 @@ func LoadConfig(path string) (*Config, error) {
 		result.Vault.Retry.NumRetries = 0
 	}
 
+	if disableIdleConnsEnv := os.Getenv(DisableIdleConnsEnv); disableIdleConnsEnv != "" {
+		result.DisableIdleConns = disableIdleConnsEnv
+	}
+
+	if result.DisableIdleConns != "" {
+		diableIdleConns := strings.ToLower(result.DisableIdleConns)
+		if strings.Contains(diableIdleConns, "auto-auth") {
```

Comment:
this might result in users abusing our lack of checking the syntax of this parameter, i.e., this allows them to give us `"auto-auth caching"` or `auto-auth;caching` or `auto-authcaching`. I worry that this sets a bad precedent for other list parameters.

Reply:
This was switched to a slice, so it shouldn't matter anymore. 😄